### ISSUE TITLE
ResourceWarning: unclosed socket when password is invalid

### ISSUE THREAD

**Issue by petronny** [oracle/python-oracledb#211]
1. What versions are you using?

oracledb 1.3.2
python 3.11
oracle db 19c

2. Is it an error or a hang or a crash?

Warning

4. What error(s) or behavior you are seeing?

sockets to connections that cannot be open (e.g. [password is invalid](https://github.com/django/django/blob/3434dbd39d373df7193ad006b970c09c1a909ea3/tests/backends/oracle/tests.py#L109-L117)) are closed automatically:

* Running `python -Wall runtests.py backends.oracle.tests.TransactionalTests` with `cx-Oracle==8.3.0`
```
$ python -Wall runtests.py backends.oracle.tests.TransactionalTests
Testing against Django installed in '/django/django' with up to 8 processes
Found 2 test(s).
Creating test database for alias 'default'...
Creating test user...
System check identified no issues (0 silenced).
..
----------------------------------------------------------------------
Ran 2 tests in 2.048s

OK
Destroying test database for alias 'default'...
Destroying test user...
Destroying test database tables...
```
* Running `python -Wall runtests.py backends.oracle.tests.TransactionalTests` with `oracledb==1.3.2`
```
python runtests.py backends.oracle.tests.TransactionalTests
Testing against Django installed in '/django/django' with up to 8 processes
Found 2 test(s).
Creating test database for alias 'default'...
Creating test user...
System check identified no issues (0 silenced).
.Exception ignored in: <socket.socket fd=3, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 38016), raddr=('127.0.0.1', 1521)>
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/traceback.py", line 236, in clear_frames
    tb.tb_frame.clear()
ResourceWarning: unclosed <socket.socket fd=3, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 38016), raddr=('127.0.0.1', 1521)>
.
----------------------------------------------------------------------
Ran 2 tests in 1.860s

OK
Destroying test database for alias 'default'...
Destroying test user...
Destroying test database tables...
```

5. Does your application call init_oracle_client()?

No.

6. Include a runnable Python script that shows the problem.

See above. Please refer to the comments in [django #15841](https://github.com/django/django/pull/15841?w=1#issuecomment-1665423324) for more information.

**Comment by anthony-tuininga** [comment]
I have pushed a patch that should correct this issue. If you are able to [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) you can verify that it corrects your issue as well.

**Comment by anthony-tuininga** [comment]
This has been included in python-oracledb 1.4.0 which was just released.
